torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -967,6 +1031,68 @@ RegisterOperators reg(
              return 0;
          },
          aliasAnalysisSpecialCase()),
+     Operator(
+         prim::ToList,
+         [](const Node* node) -> Operation {
+           return [](Stack& stack) {
+             int elem_ty_val;
+             int dim_val;
+             at::Tensor t;
+
+             pop(stack, elem_ty_val);
+             pop(stack, dim_val);
+             pop(stack, t);
+
+             // Rebuild the output type using elem_ty_val and dim_val. Start
+             // with the element type corresponding to elem_ty_val.
+             TypePtr out_ty;
+             if (elem_ty_val == 0) {
+               out_ty = IntType::get();
+             } else if (elem_ty_val == 1) {
+               out_ty = FloatType::get();
+             } else if (elem_ty_val == 2) {
+               out_ty = BoolType::get();
+             } else {
+               TORCH_INTERNAL_ASSERT(false, "Unsupported data type for tolist");
+             }
+
+             // Wrap out_ty in a ListType dim times.
+             for (int i = 0; i < dim_val; ++i) {
+               out_ty = ListType::create(out_ty);
+             }
+
+             if (auto list_type = out_ty->cast<ListType>()) {
```

Comment:
Could you add error handling for when the tensor isn't the expected type? E.g. user said List[float] but it's a non-float tensor, or it's a tensor of the wrong dims.

Reply:
Done.